test/spec.yml

```diff
@@ -112,6 +112,12 @@
             Deployed function to *
               (use "bn invoke drybeef" to invoke the function)
 
+- test: Test feedback (good-path)
```

Comment:
1. How do we filter out all the test messages that this generates?
2. In order to run this, Jenkins has to contact some frontend.  So this also transitively tests availability of that frontend, which manifests as CLI tests breaking.
3. This doesn't test the command's intended interactive behaviour.  (So far bn has been non-interactive, so unsurprisingly we have no way of testing interactive behaviour.)

Reply:
1. We will have to implement a way to identify the feedback messages and prevent them from being sent/stored. At first glance, this should be doable using the api key that is being used to run these tests. As an alternative, it could also be created such that instead of being sent to the true endpoints for slack and whatever datastore, that it is instead sent to a mock (determined via environment variable), the drawback being it's no longer an e2e test. Thoughts on those options?

2. The 2nd preventative measure for filtering out the test messages should negate this issue, since you would be using a mock.

3. You're correct, this doesn't test the interactivity. This isn't a solution for that, and while I agree that is important to have those comprehensive tests, that's not the issue this is tackling.

Reply:
I'd rather not add a fake feedback frontend in _this_ test -- as you say it's intended as a system test.  (To be clear: we _should_ also have a component test that hits only fakes; we just don't have one yet, its benefits are clear but modest compared to testing the CLI e2e).

As an alternative, could we flag messages to the feedback component as testing-only?  E.g., if you could flag the entire testing account as `bar-testing-foo`, then the feedback backend (feedbackend?) could handle them differently.  Or, if you included a specific string in feedback messages `baz-testing-quux-[64 bytes of random ASCII`, we could just filter them out downstream of the feedbackend.

Reply:
👍